### PR TITLE
Move more logic into swap state

### PR DESCRIPTION
**Pull request by ValarDragon**
This PR:
- renames `ss.spreadRewardGrowthGlobal` to `ss.globalSpreadRewardGrowthPerUnitLiquidity`
- Adds `globalSpreadRewardGrowth` into swapState, rather than tracking it in the core swap loop
- Adds swaps strategy into swapstate, lowering # of arguments needed for code de-duplication across swap types
- Combines debug log emission

This is all generally in the theme of trying to make it easier to get attention on the part of this code that matters.

### CODE REVIEW

File: x/concentrated-liquidity/swaps.go [hunk 1]
```diff
@@ -53,17 +53,23 @@ type SwapState struct {
...
-		tick:                     strategy.InitializeTickValue(p.GetCurrentTick()),
-		liquidity:                p.GetLiquidity(),
-		spreadRewardGrowthGlobal: sdk.ZeroDec(),
+		amountSpecifiedRemaining:                 specifiedAmount.ToDec(),
+		amountCalculated:                         sdk.ZeroDec(),
+		sqrtPrice:                                p.GetCurrentSqrtPrice(),
+		tick:                                     strategy.InitializeTickValue(p.GetCurrentTick()),
+		liquidity:                                p.GetLiquidity(),
+		globalSpreadRewardGrowthPerUnitLiquidity: sdk.ZeroDec(),
+		globalSpreadRewardGrowth:                 sdk.ZeroDec(),
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps.go:71]: Could you add comments explaining the difference and why we need to track both?

File: x/concentrated-liquidity/swaps.go [hunk 2]
```diff
@@ -374,7 +374,8 @@ func (k Keeper) computeOutAmtGivenIn(
 	}
 
 	// Add spread reward growth per share to the pool-global spread reward accumulator.
-	spreadRewardAccumulator.AddToAccumulator(sdk.NewDecCoins(sdk.NewDecCoinFromDec(tokenInMin.Denom, swapState.spreadRewardGrowthGlobal)))
+	spreadRewardGrowth := sdk.NewDecCoinFromDec(tokenInMin.Denom, swapState.globalSpreadRewardGrowthPerUnitLiquidity)
+	spreadRewardAccumulator.AddToAccumulator(sdk.NewDecCoins(spreadRewardGrowth))
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps.go:378]:
```suggestion
	spreadRewardGrowthPerUnitOfLiquidityCoin := sdk.NewDecCoinFromDec(tokenInMin.Denom, swapState.globalSpreadRewardGrowthPerUnitLiquidity)
	spreadRewardAccumulator.AddToAccumulator(sdk.NewDecCoins(spreadRewardGrowthPerUnitOfLiquidityCoin))
```

Comment (p0mvn) [reply x/concentrated-liquidity/swaps.go:378]: Another option is to collapse this addition into a single line to be consistent with swap in given out

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/export_test.go [hunk 3]
```diff
@@ -227,12 +227,17 @@ func (ss *SwapState) SetLiquidity(liquidity sdk.Dec) {
 	ss.liquidity = liquidity
 }
 
-func (ss *SwapState) SetSpreadRewardGrowthGlobal(spreadRewardGrowthGlobal sdk.Dec) {
-	ss.spreadRewardGrowthGlobal = spreadRewardGrowthGlobal
+// TODO: Refactor tests to ge this deleted?
```
Comment (mattverse) [inline x/concentrated-liquidity/export_test.go:230]:
```suggestion
// TODO: Refactor tests to get this deleted?
```

**Review by mattverse — COMMENTED** [review]
What's the reason we are adding `globalSpreadRewardGrowthPerUnitLiquidity` to swap state?

**Comment by ValarDragon** [comment]
globalSpreadRewardGrowthPerUnitLiquidity was already there!

The new addition is `globalSpreadRewardGrowth`, and the reasoning is because we are already tracking it as state, just not in the function

File: x/concentrated-liquidity/swaps.go [hunk 4]
```diff
@@ -53,17 +53,23 @@ type SwapState struct {
 	// Global spread reward growth per-current swap.
 	// Initialized to zero.
 	// Updated after every swap step.
-	spreadRewardGrowthGlobal sdk.Dec
+	globalSpreadRewardGrowthPerUnitLiquidity sdk.Dec
+	// global spread reward growth
+	globalSpreadRewardGrowth sdk.Dec
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/swaps.go:58]: I agree with this rename, although we might want to carry it over to incentives in the future as well for consistency (`globalUptimeGrowth` currently represents `globalUptimeGrowthPerUnitLiquidity`)

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM!

**Pull request merged**